frontend/src/components/StudentDashboard.js

```diff
@@ -88,18 +98,65 @@ const StudentDashBoard = ({ visible }) => {
     })
       .then(() => {
         setIsStatusUpdated(true);
+
+        setTimeout(() => {
+          setIsStatusUpdated(false);
+        }, 3000);
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  };
+
+  const updateInterviewDate = (date) => {
+    axios({
+      method: "PATCH",
+      url: "http://localhost:8080/student/updateInterviewDate/" + date,
+      headers: {
+        Authorization: userInfo.jwt,
+      },
+      responseType: "json",
+    })
+      .then(() => {
+        setIsInterviewDateUpdated(true);
+
+        setTimeout(() => {
+          setIsInterviewDateUpdated(false);
+        }, 3000);
       })
       .catch((error) => {
         console.error(error);
       });
   };
 
+  const hasInternship = () => {
+    return profile.studentState == listState[1];
+  };
+
   const handleChange = ($event) => {
     setProfile({ ...profile, studentState: $event.target.value });
     updateStudentStatus();
     setIsDisabled(true);
   };
 
+  const handleChangeDate = ($event) => {
+    const value = $event.target.value;
+    const dateValues = value.split("-");
+
+    if (
+      currentDate.getFullYear() == parseInt(dateValues[0]) &&
+      parseInt(dateValues[1]) >= currentDate.getMonth() + 1 &&
```

Comment:
To change the interview data, it needs to be in the next month ? (>= currentDate.getMonth() + 1)

Reply:
the logic with month is january start at 0 so i have to add +1, example for november is going to return 10 but i have to have 11 so i have to add +1, go check on the internet